### ISSUE TITLE
Set versioning for all buckets in module

### ISSUE THREAD

**Issue by tiagovrtr** [terraform-google-modules/terraform-google-cloud-storage#256]
### TL;DR

This module could allow setting `versioning` for all buckets defined in the module, the same way the `admins` input sets the same IAM role binding to all buckets instead of mapping bucket names to values as it requires in `bucket_admins`. 

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket#nested_versioning
```

### Detailed design

This could use `var.names` to generate the intended map of `name => versioning (bool)`:

``` HCL
locals {
  bucket_versioning = var.versioning ? {for name in var.names : name => true} : var.bucket_versioning
}
```

### Additional information

This would be a breaking change as the type of `versioning` would change from `map` to `bool`, but it would be more aligned with the variable naming the rest of the module uses for per-bucket and all-bucket IAM and lifecycle rules, where `[setting]` applies to all buckets and `bucket_[setting]` requires a map of `name => setting` to apply per-bucket configurations
